triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Converter.scala

```diff
@@ -348,50 +339,39 @@ object Converter {
       } yield AnyContractId(templateId, contractId)
     })
 
-  private def toAnyTemplate(v: SValue): Either[String, SValue] = {
+  private[this] def toAnyTemplate(v: SValue): Either[String, (TypeConName, SValue)] = {
```

Comment:
can we turn this into a case class, something like `TemplateValue` and `ChoiceArg` for the other one? That should remove some the pattern matching weirdness since we can just access the fields with reasonable names.